### PR TITLE
Allow to configure boot and connect timeout

### PR DESCRIPTION
**Pull request by byroot**
The hardcoded values might not be enough for all applications.

### CODE REVIEW

File: lib/spring/client/run.rb [hunk 1]
```diff
@@ -74,7 +72,7 @@ def boot_server
         env.socket_path.unlink if env.socket_path.exist?
 
         pid     = Process.spawn(server_process_env, env.server_command, out: File::NULL)
-        timeout = Time.now + BOOT_TIMEOUT
+        timeout = Time.now + String.boot_timeout
```
Comment (jenshenny) [inline lib/spring/client/run.rb:75]:
```suggestion
        timeout = Time.now + Spring.boot_timeout
```

File: lib/spring/configuration.rb [hunk 2]
```diff
@@ -5,6 +5,12 @@ class << self
     attr_accessor :application_root
     attr_writer :quiet
 
+    attr_accessor :connect_timeout
+    @connect_timeout = 5
+
+    attr_accessor :boot_timeout
+    @boot_timeout = 20
```
Comment (jenshenny) [inline lib/spring/configuration.rb:12]: These instance variables aren't being correctly set currently (the accessors return nil)

```
`+': can't convert NilClass into an exact number (TypeError)
```
It works if you place them outside of the class methods

```
module Spring
  @connect_timeout = 5
  @boot_timeout = 20

  class << self
```

**Review by jenshenny — COMMENTED** [review]
Thanks! Other than the comment I put with the accessors, LGTM

File: lib/spring/configuration.rb (hunk 2 above)
Comment (byroot) [reply lib/spring/configuration.rb]: Yeah, sorry, just did a quick patch without testing before leaving my desk. I think it should be good now.

**Pull request merged**